lib/sprockets/erb_processor.rb

```diff
@@ -20,11 +20,14 @@ def initialize(&block)
 
     def call(input)
       engine = ::ERB.new(input[:data], nil, '<>')
+      engine.filename = input[:filename]
+
       context = input[:environment].context_class.new(input)
       klass = (class << context; self; end)
+      klass.const_set(:ENV, context.env_proxy)
```

Comment:
I think we should check to see if `ENV` is already set to env_proxy before setting the constant. This call method will get called many manny times. I don't imagine that setting constants is cheap (for performance).